MediaWikiKit/MediaWikiKit/MWKArticle.m

```diff
@@ -570,9 +570,21 @@ - (NSAttributedString*)summaryHTML {
         if (!nonEmptyParagraphsWithSelectedChildren.count) {
             continue;
         }
-        NSString* lineSeparatedParagraphs = [nonEmptyParagraphsWithSelectedChildren componentsJoinedByString:@"<br/><br/>"];
-        NSData* xpathData                 = [lineSeparatedParagraphs dataUsingEncoding:NSUTF8StringEncoding];
-        return [[NSAttributedString alloc] initWithHTMLData:xpathData site:self.site];
+        NSString* lineSeparatedParagraphs = [nonEmptyParagraphsWithSelectedChildren componentsJoinedByString:@"</p><p>"];
+        lineSeparatedParagraphs = [NSString stringWithFormat:@"<p>%@</p>", lineSeparatedParagraphs];
+        NSData* xpathData = [lineSeparatedParagraphs dataUsingEncoding:NSUTF8StringEncoding];
+
+        NSAttributedString* returnValue = [[NSAttributedString alloc] initWithHTMLData:xpathData site:self.site];
+
+        // Remove trailing line break. Prob move this to a category.
+        if (returnValue.length) {
```

Comment:
So trim doesn't work on attributed strings, and it looks like the trailing line break is coming from DTCoreText. So I moved the trailing break stripping to NSAttributedString+WMFHTMLForSite.m